src/argilla/utils/modeling.py

```diff
@@ -0,0 +1,257 @@
+        if not any(record.fields[field] for record in records):
+            return None
+        field_text = [record.fields[field] for record in records]
+        field_metrics = td.extract_metrics(text=field_text, lang=self.model, metrics=self.metrics)
+        # Drop any column containing NaNs
+        field_metrics = field_metrics.dropna(axis=1, how="any")
+        # If basic metrics is None, use all basic metrics
+        if basic_metrics is None and self.metrics is None:
+            basic_metrics = self.__basic_metrics
+            field_metrics = field_metrics.loc[:, basic_metrics]
+
+        # Concatenate field name with the metric name
+        field_metrics.columns = [f"{field}_{metric}" for metric in field_metrics.columns]
+        return field_metrics
+
+    def _extract_metrics_for_all_fields(
+        self, records: List[Union[FeedbackRecord, RemoteFeedbackRecord]], fields: Optional[Union[str, List[str]]] = None
+    ) -> pd.DataFrame:
+        """
+        Extract text descriptives metrics for all named fields from a list of feedback records
+        using the TextDescriptives library.
+        Args:
+            records (List[Union[FeedbackRecord, RemoteFeedbackRecord]]): A list of FeedbackDataset or RemoteFeedbackDataset records.
+            fields (Optional[Union[str, List[str]]]): A list of fields to extract metrics for. If None, extract metrics for all fields.
+        Returns:
+            pd.DataFrame: A dataframe containing the text descriptives metrics for each record and field.
+        """
+        # If fields is None, use all fields
+        if fields is None:
+            # Programmatically select all fields in the records
+            fields = list({key for record in records for key in record.fields.keys()})
+        # Extract all metrics for each field
+        field_metrics = {
+            field: self._extract_metrics_for_single_field(records=records, field=field) for field in fields
+        }
+        field_metrics = {field: metrics for field, metrics in field_metrics.items() if metrics is not None}
+        # If there is only one field, return the metrics for that field directly
+        if len(field_metrics) == 1:
+            return list(field_metrics.values())[0]
+        # If there are multiple fields, combine metrics for each field into a single dataframe
+        final_metrics = pd.concat(field_metrics, axis=1, keys=field_metrics.keys())
+        return final_metrics
+
+    def _clean_column_name(self, col_name: str) -> str:
+        """
+        Clean the column name of a dataframe to fit a specific regex pattern.
+
+        Args:
+            col_name (str): A column name.
+
+        Returns:
+            str: A column name that fits the regex pattern.
+        """
+        col_name = col_name.lower()  # Convert to lowercase
+        col_name = re.sub(r"[^a-z0-9_]", "_", col_name)  # Replace non-alphanumeric characters with underscores
+        return col_name
```

Comment:
Could you elaborate a bit on why this function is required? Also aren't the fields already validated against a regular expression that prevents users from defining field names with other than `^(?=.*[a-z0-9])[a-z0-9_-]+$"`?

https://github.com/argilla-io/argilla/blob/0eec3c393fac79a02e2e9714aa7582a1182c4ba7/src/argilla/client/feedback/schemas/fields.py#L43

Thanks in advance for the explanation!

Reply:
You're right, this function isn't needed. I think it was left over from a prototyping notebook. I've deleted it.

Reply:
@alvarobartt After deleting this function, I found that the validation for `FloatMetaDataProperty` fails for a specific metric that text-descriptives extracts. The column name for this is called `"text_contains_lorem ipsum"` and contains a space in the column name. Is this something that needs to be addressed in the regex in line 43 that you posted above?